unity_py/resources/collection.py

```diff
@@ -15,3 +26,128 @@ def __init__(self, id):
         self._datasets = []
```

Comment:
An add_dataset method would be useful in notebook example for constructing collection, dataset, datafile

Reply:
Agreed- plan is to add this to @mcduffie notebook for application packages